pkg/agent/variables_test.go

```diff
@@ -0,0 +1,103 @@
+package agent
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/stretchr/testify/assert"
+)
+
+func MutateMap[T any](ts []T, f func(T) T) []T {
+	for i := range ts {
+		ts[i] = f(ts[i])
+	}
+	return ts
+}
+
+func trim(t string) string { return strings.Trim(t, " ") }
```

Comment:
is the trimming actually doing anything if we're asserting `Contains`?

Reply:
Aaah, Contains is done on the array of strings once we split by ";" characters rather than the strings themselves. So the assertion is "this exact string appears in the array". Which makes the trimming important.